File: CinemaReservations/Java/CinemaAllocationsDomain/src/main/java/com/sdd/cinemaallocations/TooManyTicketsRequested.java
package com.sdd.cinemaallocations;

import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.Value;

import java.util.List;

@Value
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class TooManyTicketsRequested extends SeatsAllocated {

    public TooManyTicketsRequested(int partyRequested, List<Seat> reservedSeats) {
        super(partyRequested, reservedSeats);
    }
}

File: CinemaReservations/CSharp/CinemaAllocations.Domain/TooManyTicketsRequested.cs
using System.Collections.Generic;

namespace CinemaAllocations.Domain
{
    public class TooManyTicketsRequested : SeatsAllocated
    {
        public TooManyTicketsRequested(int partyRequested) : base(new List<Seat>(), partyRequested)
        {
        }
    }
}